core/shared/src/main/scala/com/dwolla/consul/ConsulServiceDiscoveryAlg.scala

```diff
@@ -145,25 +148,42 @@ object ConsulServiceDiscoveryAlg {
+                                                           )
+                                                           (implicit L: Local[F, Span[F]]): Resource[F, F[Vector[Uri.Authority]]] =
     Stream.unfoldEval(initialConsulIndex) { maybeIndex =>
-      // since this is a background task, it doesn't make sense to attach it to the trace that initially started it
-      import natchez.Trace.Implicits.noop
-
-      lookup[F](serviceName, consulBase, maybeIndex, longPollTimeout, client)
-        .map(_.leftMap(_.some))                               // if we successfully got values, wrap them in Some so we can unNone later
-        .handleErrorWith {
-          // TODO maybe we should introduce some kind of escalating delay here?
-          Logger[F].warn(_)("🔥 An exception occurred getting service details from Consul; retrying")
-            .as((none[Vector[Uri.Authority]], maybeIndex))    // continue successfully, but emit None so the failure can be filtered out later
-        }
-        .map(_.some)                                          // this stream will unfold forever (well, until its Resource is finalized)
+      inNewLinkedRootSpan(entryPoint) { // since this is a background task, it doesn't make sense
+        import natchez.mtl._            // to directly attach it to the trace that initially started it,
+                                        // but linking it to the new root span is helpful
+
+        lookup[F](serviceName, consulBase, maybeIndex, longPollTimeout, client)
+          .map(_.leftMap(_.some)) // if we successfully got values, wrap them in Some so we can unNone later
+          .handleErrorWith {
+            // TODO maybe we should introduce some kind of escalating delay here?
+            Logger[F].warn(_)("🔥 An exception occurred getting service details from Consul; retrying")
+              .as((none[Vector[Uri.Authority]], maybeIndex)) // continue successfully, but emit None so the failure can be filtered out later
+          }
+          .map(_.some) // this stream will unfold forever (well, until its Resource is finalized)
+      }
     }
-      .unNone                                                 // errors returned by `lookup` are emitted as None, so filter them out
+      .unNone          // errors returned by `lookup` are emitted as None, so filter them out
       .holdResource(initialValue)
       .onFinalize(Logger[F].trace(s"👋 shutting down continuallyUpdating($serviceName, …)"))
       .map(_.get)
 
+  private def inNewLinkedRootSpan[F[_] : MonadCancelThrow, A](entryPoint: EntryPoint[F])
+                                                             (fa: F[A])
+                                                             (implicit L: Local[F, Span[F]]): F[A] =
+    natchez.mtl.natchezMtlTraceForLocal
+      .kernel
+      .map(Span.Options.Defaults.withLink)
+      .flatMap {
+        entryPoint
+          .root("com.dwolla.consul.ConsulServiceDiscoveryAlg.continuallyUpdating", _)
```

Comment:
We technically don't need to create a new root span here; we could use the `Local[F, Span[F]]` to set the no-op span for the `fa` scope. But we have to break bincompat to demand the `Local[F, Span[F]]`, so we might as well also ask for the `EntryPoint[F]` and create a new root span.

Reply:
And AFAICT we really do need `Local[F, Span[F]]`. One alternative I tried, using `Trace[F].span("continuallyUpdating", Span.Options.Suppress)`, did successfully suppress the span creation from the Natchez http4s middleware, but the duration of the `continuallyUpdating` was still tied to the execution of the background requests, so doesn't accomplish the main goal.